Automattic/newspack-ads
# feat(marketplace): support creative upload

1203270678582143-as-1204470726175917

Implements the support for creative upload and the ad destination URL while creating an ad order in Woo integrated with GAM.

<img width="1520" alt="image" src="https://github.com/Automattic/newspack-ads/assets/820752/f8494212-280a-4d24-bcde-9305963cb291">

### How to test

1. Setup ad products as described in #567
2. Visit the page that have the Marketplace block
3. Enter all inputs and select an image that doesn't match the required sizes
4. Click to "Proceed to payment" and confirm it fails
5. Go back and select images that match the required sizes
6. Click "Proceed to payment" and confirm you're redirected to checkout
7. Finish checkout and confirm an order is created
8. Visit the order page in Woo and confirm it includes additional "Google Ad Manager" section and order notes as in the image above
9. Click "Go to the Ad Manager" and confirm it redirects to the order in GAM
10. Navigate the order line item(s), creative(s), and confirm it has the proper metadata and destination URL set

Here are some ad creatives that fit common sizes:

![320x50](https://github.com/Automattic/newspack-ads/assets/820752/b4503eb2-0a73-45a4-b673-c9b8ce09867a)
![320x100](https://github.com/Automattic/newspack-ads/assets/820752/f906c26d-8df1-4fb3-9234-8271f1dd18a6)
![728x90](https://github.com/Automattic/newspack-ads/assets/820752/90248534-9e18-430e-b722-b494f2623541)
![970x90](https://github.com/Automattic/newspack-ads/assets/820752/04362a2e-6efa-4d99-8115-6e57c828ed05)
![970x250](https://github.com/Automattic/newspack-ads/assets/820752/78e0baaf-1340-43b5-a15b-9c4f6d0980a5)
